sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -38,6 +40,27 @@ import org.apache.spark.sql.streaming._
 import org.apache.spark.util.{Clock, UninterruptibleThread, Utils}
 
 /**
+ * Contains metadata associated with a stream execution. This information is
+ * persisted to the offset log via the OffsetSeq metadata field. Current
+ * information contained in this object includes:
+ *
+ * @param currentEventTimeWatermarkMillis: The current eventTime watermark, used to
+ * bound the lateness of data that will processed. Time unit: milliseconds
+ * @param currentBatchTimestampMillis: The current batch processing timestamp.
+ * Time unit: milliseconds
+ */
+case class StreamExecutionMetadata(
+    var currentEventTimeWatermarkMillis: Long = 0,
```

Comment:
nit: this is pretty long! how about just `batchWatermarkMs`, and `batchTimestampMs`. Its not really current if a batch is being reprocessed.